source/mainnet/smart-contracts/tutorials/wCCD/wCCD-interacting.rst

```diff
@@ -0,0 +1,748 @@
+.. _wCCD-interacting:
+
+========================================
+Interacting with the wCCD token protocol
+========================================
+
+.. note::
+
+    Before you start with part 2 of this tutorial, make sure you have access to a testnet node,
+    installed the ``concordium-client``, created an account in the mobile wallet app
+    that is funded with some CCD, and imported the keys
+    into the ``concordium-client`` configuration. If you haven't completed any of the above steps,
+    the piggy bank tutorial :ref:`part 3 <piggy-bank-preparing>`
+    will guide you through these setup steps.
+    If you haven't completed any of the above steps, you can continue
+    with :ref:`part 3 <wCCD-front-end-set-up>` of this tutorial. However, we recommend completing every part of this tutorial.
+
+Non-state-mutative functions
+----------------------------
+
+The protocol has four non-state-mutative functions (``balanceOf``, ``operatorOf``, ``tokenMetadata``, and ``supports``)
+that you can invoke on the ``proxy`` contract or the ``implementation`` contract.
+The schema is embedded into the ``implementation`` contract while you will need to provide a schema file
+when invoking the non-state-mutative functions on the ``proxy`` contract. All used schema files in this tutorial can be `downloaded <https://github.com/Concordium/concordium.github.io/tree/main/source/mainnet/smart-contracts/tutorials/wCCD/schemas>`_
+or you can create them as described in the
+comments of the `upgradable wCCD smart contract <https://github.com/Concordium/concordium-rust-smart-contracts/pull/128>`_.
+Because you will use a ``schema`` (embedded or provided with an extra flag),
+the input parameters can be provided with the ``--parameter-json`` flag.
+
+.. note::
+
+    While for testings it can be convenient to invoke these functions on the ``implementation`` contract,
+    you should always invoke them on the ``proxy`` contract in production.
```

Comment:
I applied it partly

    While testing, it can be convenient to invoke these functions on the ``implementation`` contract,
    but you should always invoke them on the ``proxy`` contract in production.